File: python/chapter_stack_and_queue/array_stack.py
class ArrayStakc:
    # 初始化栈
    def __init__(self):
        self._stack:list[int]=[]
    #获取栈的大小
    def size(self)->int:
        return len(self._stack)
    #判空
    def is_empty(self)->bool:
        return self._stack==[]
    #入栈
    def push(self,val:int):
        self._stack.append(val)
    #出栈
    def pop(self)->int:
        if self.is_empty():
            raise Exception("Stack is empty")
        return self._stack.pop()  
    #获取栈顶元素
    def peek(self)->int:
        if self.is_empty():
            raise Exception("Stack is empty")
        return self._stack[-1]
    #打印
    def to_list(self)->list[int]:
        return self._stack      

File: cpp/Study/chapter_stack_and_queue/array_stack.cpp

#include "../utils/common.hpp"

/* 基于数组实现的栈 */
class ArrayStack{
    private:
        vector <int> stack;
    public:
        //获取栈的长度
        int size(){
            return stack.size();
        }
        //判断栈是否为空
        bool isEmpty(){
            return stack.empty();
        }
        //入栈
        void push(int val){
            stack.push_back(val);
        }
        //获取栈顶元素
        int top(){
            if(isEmpty()){
                cout<<"栈为空，无法获取栈顶元素"<<endl;
                return -1;
            }
            return stack.back();
        }
        //出栈
        int pop(){
            if(isEmpty()){
                cout<<"栈为空，无法出栈"<<endl;
                return -1;
            }
            int num=top();
            stack.pop_back();
            return num;
        }
        //返回Vector
        vector<int> toArray(){
            return stack;
        }

};